arquin/compiler/converters.py

```diff
@@ -1,98 +1,93 @@
 import subprocess
-
+import numpy as np
 from qiskit.converters import circuit_to_dag, dag_to_circuit
 
-
-def edges_to_source_graph(n_vertices, edges):
-    adjacency = {vertex_idx: {} for vertex_idx in range(n_vertices)}
-    distinct_edges = set()
+def edges_to_source_graph(edges, vertex_weights):
+    '''
+    Edges of an undirected graph
+    '''
+    adjacency = {}
     for edge in edges:
-        u, v = edge
-        distinct_edges.add(tuple(sorted(edge)))
-        if v not in adjacency[u]:
-            adjacency[u][v] = 1
-        else:
-            adjacency[u][v] += 1
-        if u not in adjacency[v]:
-            adjacency[v][u] = 1
-        else:
-            adjacency[v][u] += 1
+        for counter in range(2):
+            vertex = edge[counter]
+            other_vertex = edge[(counter+1)%2]
+            if vertex in adjacency:
+                if other_vertex in adjacency[vertex]:
+                    adjacency[vertex][other_vertex] += 1
+                else:
+                    adjacency[vertex][other_vertex] = 1
+            else:
+                adjacency[vertex] = {other_vertex:1}
+    n_vertices = len(adjacency)
+    n_edges = len(edges)
 
-    vertex_weights = False
-    edge_weights = True
-    vertex_labels = False
+    has_vertex_weights = vertex_weights is not None
+    has_edge_weights = True
+    has_vertex_labels = False
     source_graph = {
-        0: "0\n",
-        1: "%d %d\n" % (n_vertices, 2 * len(distinct_edges)),
-        2: "0 %d%d%d\n" % (vertex_labels, edge_weights, vertex_weights),
-    }
+        0:'0\n',
+        1:'%d %d\n'%(n_vertices,2*n_edges),
+        2:'0 %d%d%d\n'%(has_vertex_labels,has_edge_weights,has_vertex_weights)
```

Comment:
we should use [f-string](https://realpython.com/python-f-strings/) formatting for this -- I can edit it after I finish this review